dataprofiler/data_readers/filepath_or_buffer.py

```diff
@@ -0,0 +1,31 @@
+from io import open, StringIO, BytesIO
+
+from . import data_utils
+
+class FileOrBufferHandler():
+    def __init__(self, stream_or_buffer, open_method='r', options=dict()):
+        self._stream_or_buffer = stream_or_buffer
+        self.open_method = open_method
+        self.options = options
+
+    def __enter__(self):
+        if (data_utils.is_stream_buffer(self._stream_or_buffer)):
+            print(self.options)
+            if 'seek' in self.options:
+                self._stream_or_buffer.seek(self.options['seek'],0)
+                return self._stream_or_buffer
+            else:
+                return self._stream_or_buffer
+        else:
+            self._file = open(self._stream_or_buffer, self.open_method)
```

Comment:
seek technically can be applied here, hence we could move the seek `if` out of the previous and to the end of this function